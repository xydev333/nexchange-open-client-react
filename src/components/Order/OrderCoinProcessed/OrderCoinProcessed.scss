.box.box {
  padding: 20px 0 20px 30px;
}

.box h5 {
  font-size: 18px;
}

.box h6 {
  font-size: 12px;
}

.box h5,
.box h6 {
  margin: 0;
  text-transform: none;
  color: $black;
}

.tooltip {
  text-align: left;
}

.media-left.media-left {
  padding-right: 30px;
}

.coin {
  position: relative;
  top: 8px;
  font-size: 40px;

  @media (min-width: $screen-sm-min) {
    top: 9px;
  }

  @media (min-width: $screen-md-min) {
    top: 11px;
  }
}
