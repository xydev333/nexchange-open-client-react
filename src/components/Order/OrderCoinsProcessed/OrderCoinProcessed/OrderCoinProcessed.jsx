import React, { Component } from 'react';
import { I18n } from 'react-i18next';
import copy from 'clipboard-copy';
import isFiatOrder from 'Utils/isFiatOrder';
import styles from './OrderCoinProcessed.scss';
import i18n from 'Src/i18n';
import MinMax from 'Components/MinMax/MinMax';

class OrderCoinProcessed extends Component {
  state = { order: this.props.order, hiddenAddress: true };

  componentDidMount() {
    this.prepareState(this.props);

    $(function() {
      $('[data-toggle="tooltip"]').tooltip();
    });
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.setState({ order: nextProps.order }, () => {
      this.prepareState(nextProps);
    });
  }

  triggerCopyTooltip = () => {
    if (process.env.NODE_ENV !== 'test') {
      $('#copy-address-to-clipboard').tooltip({
        trigger: 'click',
        placement: 'top',
      });

      $('#copy-address-to-clipboard')
        .tooltip('hide')
        .attr('data-original-title', i18n.t('order.copy'))
        .tooltip('show');

      setTimeout(() => {
        $('#copy-address-to-clipboard').tooltip('destroy');
      }, 1000);

      copy(this.props.order.deposit_address.address);
    }
  };

  prepareState = props => {
    if (props.type === 'Deposit') {
      this.setState({
        coin: props.order.pair.quote.code,
        oppositeCoin: props.order.pair.base.code,
        amount: parseFloat(props.order.amount_quote),
        address: props.order.deposit_address ? props.order.deposit_address.address : '',
        paymentId: props.order.deposit_address ? props.order.deposit_address.payment_id : '',
        destinationTag: props.order.deposit_address ? props.order.deposit_address.destination_tag : '',
        memo: props.order.deposit_address ? props.order.deposit_address.memo : '',
        order: props.order,
      });
    } else if (props.type === 'Receive') {
      this.setState({
        coin: props.order.pair.base.code,
        oppositeCoin: props.order.pair.quote.code,
        amount: parseFloat(props.order.amount_base),
        address: props.order.withdraw_address ? props.order.withdraw_address.address : '',
        paymentId: props.order.withdraw_address ? props.order.withdraw_address.payment_id : '',
        destinationTag: props.order.withdraw_address ? props.order.withdraw_address.destination_tag : '',
        memo: props.order.withdraw_address ? props.order.withdraw_address.memo : '',
        order: props.order,
      });
    }
  };

  renderRates() {
    let rates = ``;

    if (this.state.order && this.state.order.price) {
      rates += `${i18n.t('order.rates')}: \n`;
      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit') rates += `${(1 / this.state.order.price.rate).toFixed(6)} ${this.state.oppositeCoin}\n`;
      else if (this.props.type === 'Receive') rates += `${this.state.order.price.rate.toFixed(6)} ${this.state.oppositeCoin}\n`;

      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit')
        rates += `${((1 / this.state.order.price.rate) * this.state.order.price.rate_usd).toFixed(6)} USD\n`;
      else if (this.props.type === 'Receive') rates += `${this.state.order.price.rate_usd.toFixed(6)} USD\n`;

      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit') rates += `${((1 / this.state.order.price.rate) * this.state.order.price.rate_btc).toFixed(6)} BTC`;
      else if (this.props.type === 'Receive') rates += `${this.state.order.price.rate_btc.toFixed(6)} BTC`;

      if (this.state.order.user_provided_amount === 1 && this.props.type === 'Receive') {
        rates += `\n\n${i18n.t('order.fee')}: \n`;
        rates += `${this.state.order.withdrawal_fee} ${this.state.order.pair.base.code}`;
      } else if (this.state.order.user_provided_amount === 0 && this.props.type === 'Deposit') {
        rates += `\n\n${i18n.t('order.fee')}: \n`;
        rates += `${this.state.order.withdrawal_fee_quote} ${this.state.order.pair.quote.code}`;
      }
    }

    return rates;
  }

  toggle() {
    this.setState({
      hiddenAddress: !this.state.hiddenAddress
    });
  }

  collapseAddress() {
    console.log('COLLAPSE', this.state.address);
    this.setState({
      address: this.state.address.substring(0, 44) + '...'
    });
  }

  hasAddressId() {
    return (
      (this.state.paymentId != null && this.state.paymentId !== 'undefined') ||
      (this.state.destinationTag != null && this.state.destinationTag !== 'undefined') ||
      (this.state.memo != null && this.state.memo !== 'undefined')
    );
  }

  addressIsTooLong() {
    return (
      this.state.address != null && (this.state.address.length > 44 || this.hasAddressId())
      );
  }

  renderExpandButton() {
    let renderedExandButton;
    renderedExandButton = null;
    if (this.addressIsTooLong()) {
      renderedExandButton =
        <a className={`${styles['clickable-text']}`} onClick={this.toggle.bind(this)}>
          {this.state.hiddenAddress ? 'Expand' : 'Collapse'}
        </a>;
    }
    return renderedExandButton;
  }

  renderAddress() {
    let renderedAddress;
    let addressId;

    if (this.state.paymentId) {
      addressId = this.state.paymentId;
    } else if (this.state.destinationTag) {
      addressId = this.state.destinationTag;
    } else if (this.state.memo) {
      address = this.state.memo;
    } else {
      addressId = null;
    }

    if (this.addressIsTooLong() && this.state.hiddenAddress){
      renderedAddress =
        <div className={styles.address}>
          <h6>{this.addressIsTooLong() ? this.state.address.substring(0, 44) + '...' : this.state.address}</h6>
        </div>;
    } else {
      renderedAddress =
        <div className={styles.address}>
          <h6>{this.state.address}</h6>
          <h6>{addressId}</h6>
        </div>;
    }
    return renderedAddress;
  }

  render() {
    return (
      <I18n ns="translations">
        {t => (
          <div className={`col-xs-12 col-sm-6 ${styles['col-sm-6']} ${this.props.type === 'Receive' ? styles['pull-right-md'] : ''}`}>
            <div
              className={`${styles.box} box ${this.props.type === 'Deposit' && isFiatOrder(this.props.order) ? 'fiat' : ''} ${
                !isFiatOrder(this.props.order) || this.props.type === 'Receive' ? styles['crypto'] : ''
              }`}
            >
              <div className={`${styles['media-left']}`}>
                <i className={`${styles.coin} cc ${this.state.coin}`} />
              </div>

              <div className={`${styles['media-right']}`}>
                <h5>
                  {t('order.' + this.props.type)}{' '}
                  <b>
                    {this.state.amount} {this.state.coin}
                  </b>
                  <i
                    className="fa fa-question-circle"
                    data-toggle="tooltip"
                    data-placement="top"
                    style={{ marginLeft: 8 }}
                    data-original-title={this.renderRates()}
                  />
                </h5>
                {this.renderAddress()}
                {this.props.type === 'Deposit' &&
                  !isFiatOrder(this.props.order) && (
                    <i
                      id="copy-address-to-clipboard"
                      className={`${styles.copy} fas fa-copy`}
                      data-test="copy-address"
                      onClick={() => this.triggerCopyTooltip()}
                    />
                  )}
                {this.renderExpandButton()}
                <MinMax {...this.props} />
              </div>
            </div>
          </div>
        )}
      </I18n>
    );
  }
}

export default OrderCoinProcessed;
