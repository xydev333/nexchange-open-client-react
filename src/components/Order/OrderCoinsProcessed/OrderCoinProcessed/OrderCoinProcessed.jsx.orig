import React, { Component } from 'react';
<<<<<<< HEAD:src/components/Order/OrderCoinsProcessed/OrderCoinProcessed/OrderCoinProcessed.jsx
import isFiatOrder from 'Utils/isFiatOrder';
import styles from './OrderCoinProcessed.scss';
=======
import isFiatOrder from '../../helpers/isFiatOrder';
import i18n from '../../i18n';
import { I18n } from 'react-i18next';
>>>>>>> Translated further components:src/containers/order/CoinProcessed.jsx

class OrderCoinProcessed extends Component {
  state = { order: this.props.order };

  componentDidMount() {
    this.prepareState(this.props);
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.setState({ order: nextProps.order }, () => {
      this.prepareState(nextProps);
    });
  }

  prepareState = props => {
    if (props.type === 'Deposit') {
      this.setState({
        coin: props.order.pair.quote.code,
        oppositeCoin: props.order.pair.base.code,
        amount: parseFloat(props.order.amount_quote),
        address: props.order.deposit_address ? props.order.deposit_address.address : '',
        order: props.order,
      });
    } else if (props.type === 'Receive') {
      this.setState({
        coin: props.order.pair.base.code,
        oppositeCoin: props.order.pair.quote.code,
        amount: parseFloat(props.order.amount_base),
        address: props.order.withdraw_address ? props.order.withdraw_address.address : '',
        order: props.order,
      });
    }
  };

  render() {
    let rates = ``;

    if (this.state.order && this.state.order.price) {
      rates += `${i18n.t('order.rates')}: \n`;
      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit') rates += `${(1 / this.state.order.price.rate).toFixed(8)} ${this.state.oppositeCoin}\n`;
      else if (this.props.type === 'Receive') rates += `${this.state.order.price.rate.toFixed(8)} ${this.state.oppositeCoin}\n`;

      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit')
        rates += `${((1 / this.state.order.price.rate) * this.state.order.price.rate_usd).toFixed(8)} USD\n`;
      else if (this.props.type === 'Receive') rates += `${this.state.order.price.rate_usd.toFixed(8)} USD\n`;

      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit') rates += `${((1 / this.state.order.price.rate) * this.state.order.price.rate_btc).toFixed(8)} BTC`;
      else if (this.props.type === 'Receive') rates += `${this.state.order.price.rate_btc.toFixed(8)} BTC`;

      if (this.state.order.user_provided_amount === 1 && this.props.type === 'Receive') {
        rates += `\n\n${'order.fee'}: \n`;
        rates += `${this.state.order.withdrawal_fee} ${this.state.order.pair.base.code}`;
      } else if (this.state.order.user_provided_amount === 0 && this.props.type === 'Deposit') {
        rates += `\n\n${'order.fee'}: \n`;
        rates += `${this.state.order.withdrawal_fee_quote} ${this.state.order.pair.quote.code}`;
      }
    }

    return (
<<<<<<< HEAD:src/components/Order/OrderCoinsProcessed/OrderCoinProcessed/OrderCoinProcessed.jsx
      <div className={`col-xs-12 col-sm-6 ${styles['col-sm-6']} ${this.props.type === 'Receive' ? styles['pull-right-md'] : null}`}>
        <div className={`${styles.box} box media ${this.props.type === 'Deposit' && isFiatOrder(this.props.order) ? 'fiat' : ''}`}>
          <div className={`${styles['media-left']} media-left`}>
            <i className={`${styles.coin} cc ${this.state.coin}`} />
=======
    <I18n ns="translations">
     {(t) => (
      <div className="col-xs-12 col-sm-6">
        <div className={`coin-box box media ${this.props.type === 'Deposit' && isFiatOrder(this.props.order) ? 'fiat' : ''}`}>
          <div className="media-left">
            <i className={`coin-icon cc ${this.state.coin}`} />
>>>>>>> Translated further components:src/containers/order/CoinProcessed.jsx
          </div>

          <div className="media-body">
            <h5>
              {this.props.type}{' '}
              <b>
<<<<<<< HEAD:src/components/Order/OrderCoinsProcessed/OrderCoinProcessed/OrderCoinProcessed.jsx
                {this.state.amount} {this.state.coin}
=======
                {t('order.'+this.props.type)} {this.state.amount} {this.state.coin}
>>>>>>> Translated further components:src/containers/order/CoinProcessed.jsx
              </b>
              <i
                className="fa fa-question-circle"
                data-toggle="tooltip"
                data-placement="top"
                style={{ marginLeft: 8 }}
                data-original-title={rates}
              />
            </h5>
            <h6>{this.state.address}</h6>
          </div>
        </div>
      </div>
      )}
	 </I18n>
    );
  }
}

export default OrderCoinProcessed;
