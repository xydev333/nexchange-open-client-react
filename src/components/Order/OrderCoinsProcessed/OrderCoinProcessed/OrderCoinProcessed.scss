.box.box {
  padding: 20px 0 20px 10px;
  overflow: hidden;

  @media (min-width: $screen-lg-min) {
    padding-left: 30px;
  }
}

.box h5 {
  font-size: 16px;

  @media (min-width: $screen-ms-min) {
    font-size: 18px;
  }
}

.box h6 {
  font-size: 12px;
}

.box h5,
.box h6 {
  margin: 0;
  text-transform: none;
  color: $black;
}

:global .order-fiat .box.fiat h5 {
  margin: 9px 0;
}

.tooltip {
  text-align: left;
}

.media-left.media-left {
  padding-right: 10px;
  float: left;
  width: 50px;
  text-align: center;

  @media (min-width: $screen-lg-min) {
    padding-right: 30px;
    width: 70px;
  }
}

.media-right.media-right {
  padding-right: 10px;
  float: left;
  width: calc(100% - 50px);

  @media (min-width: $screen-lg-min) {
    padding-right: 15px;
    width: calc(100% - 70px);
  }
}

.address {
  float: left;
  overflow-y: auto;
  max-width: calc(100% - 30px);
  margin-right: 10px;
}

.copy {
  position: relative;
  top: -2px;
  color: $gray;
  cursor: pointer;
}

.coin {
  position: relative;
  top: 8px;
  font-size: 30px;

  @media (min-width: $screen-sm-min) {
    top: 9px;
  }

  @media (min-width: $screen-lg-min) {
    top: 11px;
    font-size: 40px;
  }
}

.col-sm-6.col-sm-6 {
  @media (min-width: $screen-md-min) {
    width: 46%;
  }
}

.pull-right-md.pull-right-md {
  @media (min-width: $screen-md-min) {
    float: right;
  }
}
