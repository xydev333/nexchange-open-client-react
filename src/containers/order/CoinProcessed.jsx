import React, { Component } from 'react';
import Helpers from '../../helpers';

class CoinProcessed extends Component {
  constructor(props) {
    super(props);
    this.state = { order: this.props.order };
    this.prepareState = this.prepareState.bind(this);
  }

  componentDidMount() {
    this.prepareState(this.props);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ order: nextProps.order }, () => {
      this.prepareState(nextProps);
    });
  }

  prepareState(props) {
    if (props.type === 'Deposit') {
      this.setState({
        coin: props.order.pair.quote.code,
        oppositeCoin: props.order.pair.base.code,
        amount: parseFloat(props.order.amount_quote),
        address: props.order.deposit_address
          ? props.order.deposit_address.address
          : '',
        order: props.order,
      });
    } else if (props.type === 'Receive') {
      this.setState({
        coin: props.order.pair.base.code,
        oppositeCoin: props.order.pair.quote.code,
        amount: parseFloat(props.order.amount_base),
        address: props.order.withdraw_address
          ? props.order.withdraw_address.address
          : '',
        order: props.order,
      });
    }
  }

  render() {
    let rates = ``;

    if (this.state.order && this.state.order.price) {
      rates += `Rates at order creation: \n`;
      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit')
        rates += `${(1 / this.state.order.price.rate).toFixed(8)} ${
          this.state.oppositeCoin
        }\n`;
      else if (this.props.type === 'Receive')
        rates += `${this.state.order.price.rate.toFixed(8)} ${
          this.state.oppositeCoin
        }\n`;

      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit')
        rates += `${(
          1 /
          this.state.order.price.rate *
          this.state.order.price.rate_usd
        ).toFixed(8)} USD\n`;
      else if (this.props.type === 'Receive')
        rates += `${this.state.order.price.rate_usd.toFixed(8)} USD\n`;

      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit')
        rates += `${(
          1 /
          this.state.order.price.rate *
          this.state.order.price.rate_btc
        ).toFixed(8)} BTC`;
      else if (this.props.type === 'Receive')
        rates += `${this.state.order.price.rate_btc.toFixed(8)} BTC`;

      if (
        this.state.order.user_provided_amount === 1 &&
        this.props.type === 'Receive'
      ) {
        rates += `\n\nWithdrawal fee: \n`;
        rates += `${this.state.order.withdrawal_fee} ${
          this.state.order.pair.base.code
        }`;
      } else if (
        this.state.order.user_provided_amount === 0 &&
        this.props.type === 'Deposit'
      ) {
        rates += `\n\nWithdrawal fee: \n`;
        rates += `${this.state.order.withdrawal_fee_quote} ${
          this.state.order.pair.quote.code
        }`;
      }
    }

    return (
      <div className="col-xs-12 col-sm-6">
        <div
          className={`coin-box box media ${
            this.props.type === 'Deposit' &&
            Helpers.isFiatOrder(this.props.order)
              ? 'fiat'
              : ''
          }`}
        >
          <div className="media-left">
            <i
              className={`coin-icon cc-${this.state.coin} ${this.state.coin}`}
            />
          </div>

          <div className="media-body">
            <h5>
              <b>
                {this.props.type} {this.state.amount} {this.state.coin}
              </b>

              <i
                className="fa fa-question-circle"
                data-toggle="tooltip"
                data-placement="top"
                style={{ marginLeft: 8 }}
                data-original-title={rates}
              />
            </h5>
            <h6>{this.state.address}</h6>
          </div>
        </div>
      </div>
    );
  }
}

export default CoinProcessed;
