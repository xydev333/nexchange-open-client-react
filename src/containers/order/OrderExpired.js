import React, { Component } from 'react';

class OrderExpired extends Component {
	componentDidMount() {
		$('[data-toggle="tooltip"], [rel="tooltip"]').tooltip();
	}

	render() {
	    return (
			<div className="text-center">
				<h2>Order Expired</h2>
				<h5>Your order is expired. You can make a new exchange order.</h5>
				
				<div>
					<h4
						className="text-warning"
						data-toggle="tooltip"
						data-placement="top"
						data-original-title="If you have already sent your coins, donâ€™t worry. Once we get them, the status of the order will be updated retroactively.">
						Already sent?
					</h4>
				</div>
			</div>
	    );
	}
}

export default OrderExpired;
