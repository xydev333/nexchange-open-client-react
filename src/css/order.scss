@import "variables";
@import "mixins";
@import "bootstrap/variables";
@import "bootstrap/mixins";

#order { background: #f8f7f8; padding: 15px 0 0 0;
	a { color: $green; 
		&:visited, &:focus { text-decoration: none; }
		&:hover { text-decoration: underline; }
		&.btn { color: #fff;
			&:hover { text-decoration: none; }
		}
	}
	#order-ref { margin-bottom: 15px; }
	.box {
		h4 { font-size: 15px; }
		h5, h6 { margin: 0; text-transform: none; }
		.coin-icon { position: relative; top: 8px; font-size: 28px;
			@media (min-width: $screen-sm-min) { top: 9px; }
			@media (min-width: $screen-md-min) { top: 11px; }
		}
		.text-warning { color: $green; }		

		@media (min-width: $screen-sm-min) { margin-bottom: 30px; }
	}

	&.order-fiat {
		.box.fiat {
			h5 { margin-top: 11px; padding-bottom: 7px;
				@media (min-width: $screen-sm-min) { margin-top: 11px; padding-bottom: 13px; }
			}
		}
	}

	.coin-box { padding-left: 10px;
		.tooltip-inner { text-align: left; }
	}
	#time-remaining { color: #2cb4a0; }
	.order-status-section { 
		h2 { margin-top: 0; }
		.explanation { font-size: 16px; margin-top: 15px; }
	}


	@media (min-width: $screen-sm-min) { padding: 25px 0;
		h5 { font-size: 15px; }
		h6 { font-size: 12px; }
		.box h4 { font-size: 18px; }
	}

	@media (min-width: $screen-md-min) {
		h5 { font-size: 17px; }
		h6 { font-size: 16px; }
	}
}

#order-header {
	#bookmark-button { font-size: 20px; padding: 0; margin-top: 0; color: $green; }

	@media (min-width: $screen-ms-min) {
		h3 { float: left; }
		#bookmark-button { float: right; margin-top: 22px; }
	}
}

#order-payment { text-align: center;
	h3 { margin-top: 0; }
	@media (min-width: $screen-ms-min) { text-align: left; }
}

#bookmark {
	.bookmark-text{ font-size: 19px; }
	.input-group-addon { cursor: pointer; padding-top: 0;
		i { font-size: 18px; }
	}
	.form-group { margin-top: 10px; }
	input { font-size: 18px; }
}

.order-status-section {
	h2 { font-size: 24px; }

	@media (min-width: $screen-sm-min) {
		h2 { font-size: 30px; }
	}
}

#share-referral, #notifications { text-align: center;
	.box { padding: 20px; }
	.share { margin-top: 15px; }
	.fa { font-size: 45px; margin: 0 15px 0 0; color: #d6d6d6; }
	h2 { margin-top: 0; font-size: 22px;
		@media (min-width: $screen-sm-min) { font-size: 26px; }
		@media (min-width: $screen-md-min) { font-size: 34px; }
	}
}

#notifications {
	.form-group { margin-top: 10px; }
	button { width: 100%; margin-bottom: 0; }
	form { padding-top: 15px;
		h4 { position: absolute; top: -15px; left: 50%; transform: translate(-50%, 0); }
	}
	@media (min-width: $screen-sm-min) { button { width: auto; } }
}

@include keyframes(color-change) {
	from { color: #adadad; }
	to { color: #fbc02d; } 
}

@import "order-status";
@import "referral";
