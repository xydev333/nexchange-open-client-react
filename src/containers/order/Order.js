import React, { Component } from 'react';
import { connect } from 'react-redux';

import { fetchOrder } from '../../actions';

import isFiatOrder from '../../helpers/isFiatOrder';
import config from '../../config';

import '../../css/order.scss';

import OrderInfo from './OrderInfo';
import OrderTop from './OrderTop';
import CoinProcessed from './CoinProcessed';

import NotFound from '../../components/NotFound';
import Loading from '../../components/Loading';

import Notifications from '../../containers/Notifications';
import RefundAddress from '../../containers/RefundAddress';
import ReferralBox from '../../containers/ReferralBox';

class Order extends Component {
  constructor(props) {
    super(props);

    if (this.props.order && this.props.match.params.orderRef === this.props.order.unique_reference) {
      this.state = { order: this.props.order };
    } else {
      this.state = {};
    }
  }

  componentDidMount() {
    this.props.fetchOrder(this.props.match.params.orderRef);
  }

  componentDidUpdate(prevProps) {
    if (this.props.location !== prevProps.location) {
      clearTimeout(this.timeout);
      clearInterval(this.interval);
      this.props.fetchOrder(this.props.match.params.orderRef);
    }
  }

  componentWillUnmount() {
    clearInterval(this.interval);
    clearTimeout(this.timeout);
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.setState({ order: nextProps.order });

    this.timeout = setTimeout(() => {
      this.props.fetchOrder(this.props.match.params.orderRef);
    }, config.ORDER_DETAILS_FETCH_INTERVAL);

    if (nextProps.order !== 429) {
      this.setState({ order: nextProps.order });

      if (
        this.props.order &&
        this.props.order.status_name.length > 0 &&
        this.props.order.status_name[0][0] === 11 &&
        nextProps.order.status_name[0][0] === 12
      ) {
        window.ga('send', 'event', 'Order', 'order paid', nextProps.unique_reference);
      }
    }
  }

  render() {
    if (this.state.order == null) {
      return <Loading />;
    } else if (this.state.order === 404) {
      return <NotFound />;
    } else if (typeof this.state.order === 'object') {
      return (
        <div id="order" className={isFiatOrder(this.state.order) ? 'order-fiat' : 'order-crypto'}>
          <div className="container">
            <OrderTop order={this.state.order} />

            <div className="row">
              <CoinProcessed type="Deposit" order={this.state.order} />
              <CoinProcessed type="Receive" order={this.state.order} />

              <OrderInfo order={this.state.order} />
              <Notifications order={this.state.order} />

              {!isFiatOrder(this.state.order) && <RefundAddress order={this.state.order} />}

              <ReferralBox order={this.state.order} />
            </div>
          </div>
        </div>
      );
    }
  }
}

const mapStateToProps = ({ order }) => ({ order });

export default connect(mapStateToProps, { fetchOrder })(Order);
