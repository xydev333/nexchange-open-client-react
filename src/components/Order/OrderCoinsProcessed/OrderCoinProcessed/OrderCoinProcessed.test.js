import React from 'react';
import { shallow, mount } from 'enzyme';
import OrderCoinProcessed from './OrderCoinProcessed';
import order from 'Mocks/order';
import orderFiat from 'Mocks/orderFiat';

describe('OrderCoinProcessed', () => {
  it('renders correctly and updates props (deposit and crypto)', () => {
    const wrapShallow = shallow(<OrderCoinProcessed type="Deposit" order={order} />);
    expect(wrapShallow).toMatchSnapshot();
    wrapShallow.setProps({ order });
    expect(wrapShallow).toMatchSnapshot();
  });

  it('renders correctly and updates props (receive and crypto)', () => {
    const wrapShallow = shallow(<OrderCoinProcessed type="Receive" order={order} />);
    expect(wrapShallow).toMatchSnapshot();
    wrapShallow.setProps({ order });
    expect(wrapShallow).toMatchSnapshot();
  });

  it('renders correctly (deposit and fiat)', () => {
    const wrapShallow = shallow(<OrderCoinProcessed type="Deposit" order={orderFiat} />);
    expect(wrapShallow).toMatchSnapshot();
  });

  it('renders correctly (receive and fiat)', () => {
    const wrapShallow = shallow(<OrderCoinProcessed type="Receive" order={orderFiat} />);
    expect(wrapShallow).toMatchSnapshot();
  });

  it('copy trigger is called', () => {
    const spy = jest.spyOn(OrderCoinProcessed.prototype, 'triggerCopyTooltip');
    const wrapShallow = mount(<OrderCoinProcessed type="Deposit" order={order} />);

    expect(spy).not.toHaveBeenCalled();
    wrapShallow.find('[data-test="copy-address"]').simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
