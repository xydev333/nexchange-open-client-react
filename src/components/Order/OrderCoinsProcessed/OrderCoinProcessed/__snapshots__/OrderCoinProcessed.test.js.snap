// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderCoinProcessed renders correctly (deposit and fiat) 1`] = `ShallowWrapper {}`;

exports[`OrderCoinProcessed renders correctly (receive and fiat) 1`] = `ShallowWrapper {}`;

exports[`OrderCoinProcessed renders correctly and passes payment id to order initial 1`] = `ShallowWrapper {}`;

exports[`OrderCoinProcessed renders correctly and passes payment id to order initial 2`] = `ShallowWrapper {}`;

exports[`OrderCoinProcessed renders correctly and updates props (deposit and crypto) 1`] = `ShallowWrapper {}`;

exports[`OrderCoinProcessed renders correctly and updates props (deposit and crypto) 2`] = `ShallowWrapper {}`;

exports[`OrderCoinProcessed renders correctly and updates props (receive and crypto) 1`] = `ShallowWrapper {}`;

exports[`OrderCoinProcessed renders correctly and updates props (receive and crypto) 2`] = `ShallowWrapper {}`;
